### PR TITLE
Improve discovery of Java JDK for mock plugins

### PR DESCRIPTION
**Pull request by novotnyr**
When discovering JDK for mock plugins and their descriptors, investigate

1. `pluginverifier.testjava.home` system property
2. `JAVA_HOME` environment variable
3. SDKMan common location (MacOS)
4. `java.home` system property 
5. fixed paths (Linux).

### CODE REVIEW

File: intellij-plugin-verifier/verifier-test/src/test/java/com/jetbrains/pluginverifier/tests/mocks/TestJdkDescriptorProvider.kt [hunk 1]
```diff
@@ -4,27 +4,60 @@ import com.jetbrains.plugin.structure.base.utils.exists
...
-    val jvmHomeDir = Paths.get(JVM_HOME_DIRS)
-    if (jvmHomeDir.exists()) {
-      val someJdk = jvmHomeDir.listFiles().firstOrNull { it.isDirectory }
-      if (someJdk != null) {
-        println("Using $someJdk as JDK in tests")
-        return someJdk
+    return candidates.filterNotNull()
+      .firstOrNull(Path::exists)
+      .also {
+        println("Using $it as JDK in tests")
```
Comment (YannCebron) [inline intellij-plugin-verifier/verifier-test/src/test/java/com/jetbrains/pluginverifier/tests/mocks/TestJdkDescriptorProvider.kt:30]: this becomes a bit noisy, can we log it with debug level instead?

File: intellij-plugin-verifier/verifier-test/build.gradle.kts [hunk 2]
```diff
@@ -6,6 +6,10 @@ dependencies {
 
   //bytecode generation library
   implementation(libs.byteBuddy)
+
```
Comment (YannCebron) [inline intellij-plugin-verifier/verifier-test/build.gradle.kts:9]: this change looks unrelated

**Review by YannCebron — CHANGES_REQUESTED** [review]

File: intellij-plugin-verifier/verifier-test/build.gradle.kts (hunk 2 above)
Comment (novotnyr) [reply intellij-plugin-verifier/verifier-test/build.gradle.kts:8]: Dependencies were put into catalog.

File: intellij-plugin-verifier/verifier-test/src/test/java/com/jetbrains/pluginverifier/tests/mocks/TestJdkDescriptorProvider.kt [hunk 3]
```diff
@@ -27,7 +30,7 @@ object TestJdkDescriptorProvider {
     return candidates.filterNotNull()
       .firstOrNull(Path::exists)
       .also {
-        println("Using $it as JDK in tests")
+        LOG.info("Using $it as JDK in tests")
```
Comment (YannCebron) [inline intellij-plugin-verifier/verifier-test/src/test/java/com/jetbrains/pluginverifier/tests/mocks/TestJdkDescriptorProvider.kt:33]: INFO gets logged by default for me :) quite noisy, still

**Review by YannCebron — APPROVED** [review]
